src/LightInject/LightInject.cs

```diff
@@ -6537,8 +6543,17 @@ private void BuildImplementationMap(Type implementingType, IServiceRegistry serv
         private void RegisterInternal(Type serviceType, Type implementingType, IServiceRegistry serviceRegistry, ILifetime lifetime)
         {
             var serviceTypeInfo = serviceType.GetTypeInfo();
+
+            
             if (serviceTypeInfo.IsGenericType && serviceTypeInfo.ContainsGenericParameters)
             {
+                var implementingTypeInfo = implementingType.GetTypeInfo();
+
+                if (implementingTypeInfo.IsGenericType && implementingTypeInfo.ContainsGenericParameters && implementingTypeInfo.GenericTypeParameters.Length > serviceTypeInfo.GenericTypeArguments.Length)
```

Comment:
I see your point. I will add a new test with your example and find a better verification.